addons/account/controllers/portal.py

```diff
@@ -127,11 +127,12 @@ def portal_my_invoice_detail(self, invoice_id, access_token=None, report_type=No
     # My Home
     # ------------------------------------------------------------
 
-    def details_form_validate(self, data):
+    def details_form_validate(self, data, partner_creation=False):
         error, error_message = super(PortalAccount, self).details_form_validate(data)
         # prevent VAT/name change if invoices exist
         partner = request.env['res.users'].browse(request.uid).partner_id
-        if not partner.can_edit_vat():
+        # This can be called without user, in which case this check can be skipped.
+        if (partner_creation and request.env.user._is_public()) and not partner.can_edit_vat():
```

Comment:
the boolean condition seems reversed, that would disable the vat check for the normal edition case, no? (need to block when `not partner_creation`)

also, do you really need to check for the public user? I don't think that's necessary if we know we're creating a new partner anyway

```suggestion
        if not partner_creation and not partner.can_edit_vat():
```